clang/test/SemaSYCL/allow-all-features-in-constexpr.cpp

```diff
@@ -0,0 +1,129 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -fcxx-exceptions -verify -fsyntax-only -std=c++23 -triple spir64 -aux-triple x86_64 -fsycl-allow-all-features-in-constexpr %s
+
+// The test checks that all SYCL device code limitations are lifted in
+// manifestly constant-evaluated expressions under an option.
+#include "sycl.hpp"
+
+struct Base {
+  virtual constexpr void f() {
+  }
+};
+
+struct Derived : Base {
+  constexpr void f() override {}
+};
+
+constexpr int recurse(int x) {
+  if (x > 1)
+    return 1 + recurse(x-1);
+  return x;
+}
+
+
+constexpr int variadic(int p, ...) {
+  // Usage of these is not yet allowed in constexpr in C++, still check that
+  // variadics are allowed in device code.
+  // __builtin_va_list args;
```

Comment:
What are these commented built-ins for?

Reply:
These are leftovers from various testing. I'll remove them now.
I wanted to check if I can use these as well, but it seems there is no C++ version that allows using these in constant expressions.